Jolokia2 paniced from nil reponse

### Relevant telegraf.conf

```toml
[[inputs.jolokia2_agent]]
  interval="10s"
  urls = ["http://localhost:8081/jolokia","http://localhost:7777/jolokia"]

  [[inputs.jolokia2_agent.metric]]
    name  = "java_garbage_collector"
    mbean = "java.lang:name=*,type=GarbageCollector"
    paths = ["CollectionTime","CollectionCount"]

  [[inputs.jolokia2_agent.metric]]
    name  = "java_last_garbage_collection"
    mbean = "java.lang:name=*,type=GarbageCollector"
    paths = ["LastGcInfo/duration","LastGcInfo/GcThreadCount"]

[[outputs.file]]
  files = ["stdout"]
```


### Logs from Telegraf

```text
panic: There should be a map here for java.lang:name=*,type=GarbageCollector!


goroutine 57 [running]:
github.com/influxdata/telegraf/plugins/inputs/jolokia2/common.(*pointBuilder).Build(0xc000f4acc1?, {0xc000f4acc1, 0x26}, {0x0, 0x0})
        /Users/papapiya/go/src/github.com/influxdata/telegraf/plugins/inputs/jolokia2/common/point_builder.go:38 +0x31f
github.com/influxdata/telegraf/plugins/inputs/jolokia2/common.(*Gatherer).generatePoints(0xc000034300?, {{0xc000f4ac31, 0x1c}, {0xc000f4acc1, 0x26}, {0xc0006292e0, 0x2, 0x2}, {0x0, 0x0}, ...}, ...)
        /Users/papapiya/go/src/github.com/influxdata/telegraf/plugins/inputs/jolokia2/common/gatherer.go:97 +0x528
github.com/influxdata/telegraf/plugins/inputs/jolokia2/common.(*Gatherer).gatherResponses(0xc0007f8180, {0xc000034300, 0x5, 0x8}, 0x0?, {0x9e44170, 0xc0006296e0})
        /Users/papapiya/go/src/github.com/influxdata/telegraf/plugins/inputs/jolokia2/common/gatherer.go:57 +0x4f8
github.com/influxdata/telegraf/plugins/inputs/jolokia2/common.(*Gatherer).Gather(0xc0007f8180, 0xc0006081c0, {0x9e44170, 0xc0006296e0})
        /Users/papapiya/go/src/github.com/influxdata/telegraf/plugins/inputs/jolokia2/common/gatherer.go:42 +0x24c
github.com/influxdata/telegraf/plugins/inputs/jolokia2/jolokia2_agent.(*JolokiaAgent).Gather.func1(0xc0006081c0)
        /Users/papapiya/go/src/github.com/influxdata/telegraf/plugins/inputs/jolokia2/jolokia2_agent/jolokia2_agent.go:67 +0x97
created by github.com/influxdata/telegraf/plugins/inputs/jolokia2/jolokia2_agent.(*JolokiaAgent).Gather
        /Users/papapiya/go/src/github.com/influxdata/telegraf/plugins/inputs/jolokia2/jolokia2_agent/jolokia2_agent.go:64 +0x1ec
```


### System info

Telegraf 1.23.0, Darwin 17.5.0

### Docker

_No response_

### Steps to reproduce

1.Use relevant config to start telegraf: `./telegraf --config telegraf.conf`
2.It comes panic info in stdout and stop.
3.Jolokia request and response list here:
* request:
```json
[
    {
        "attribute": [
            "CollectionCount",
            "CollectionTime"
        ],
        "mbean": "java.lang:name=*,type=GarbageCollector",
        "type": "read"
    },
    {
        "attribute": "LastGcInfo",
        "mbean": "java.lang:name=*,type=GarbageCollector",
        "path": "duration",
        "type": "read"
    },
    {
        "attribute": "LastGcInfo",
        "mbean": "java.lang:name=*,type=GarbageCollector",
        "path": "GcThreadCount",
        "type": "read"
    }
]
```
* response
```json
[
    {
        "request": {
            "attribute": [
                "CollectionCount",
                "CollectionTime"
            ],
            "mbean": "java.lang:name=*,type=GarbageCollector",
            "type": "read"
        },
        "status": 200,
        "timestamp": 1656469537,
        "value": {
            "java.lang:name=ConcurrentMarkSweep,type=GarbageCollector": {
                "CollectionCount": 3,
                "CollectionTime": 683
            },
            "java.lang:name=ParNew,type=GarbageCollector": {
                "CollectionCount": 443,
                "CollectionTime": 29926
            }
        }
    },
    {
        "request": {
            "attribute": "LastGcInfo",
            "mbean": "java.lang:name=*,type=GarbageCollector",
            "path": "duration",
            "type": "read"
        },
        "status": 200,
        "timestamp": 1656469537,
        "value": null
    },
    {
        "request": {
            "attribute": "LastGcInfo",
            "mbean": "java.lang:name=*,type=GarbageCollector",
            "path": "GcThreadCount",
            "type": "read"
        },
        "status": 200,
        "timestamp": 1656469537,
        "value": null
    }
]
```

### Expected behavior

When jolokia response with null value, the telegraf process not panic and exit.

### Actual behavior

Telegraf paniced with null response value.

### Additional info

When mbean contains "*" and paths contains subpath like "LastGcInfo/duration", the jolokia response with  null value

--- a/plugins/inputs/jolokia2/common/gatherer.go
+++ b/plugins/inputs/jolokia2/common/gatherer.go
@@ -94,7 +94,13 @@ func (g *Gatherer) generatePoints(metric Metric, responses []ReadResponse) ([]po
 		}
 
 		pb := NewPointBuilder(metric, response.RequestAttributes, response.RequestPath)
-		for _, point := range pb.Build(metric.Mbean, response.Value) {
+		ps, err := pb.Build(metric.Mbean, response.Value)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		for _, point := range ps {
 			if response.RequestTarget != "" {
 				point.Tags["jolokia_agent_url"] = response.RequestTarget
 			}

--- a/plugins/inputs/jolokia2/common/point_builder.go
+++ b/plugins/inputs/jolokia2/common/point_builder.go
@@ -27,15 +27,15 @@ func NewPointBuilder(metric Metric, attributes []string, path string) *pointBuil
 }
 
 // Build generates a point for a given mbean name/pattern and value object.
-func (pb *pointBuilder) Build(mbean string, value interface{}) []point {
+func (pb *pointBuilder) Build(mbean string, value interface{}) ([]point, error) {
 	hasPattern := strings.Contains(mbean, "*")
-	if !hasPattern {
+	if !hasPattern || value == nil {
 		value = map[string]interface{}{mbean: value}
 	}
 
 	valueMap, ok := value.(map[string]interface{})
-	if !ok { // FIXME: log it and move on.
-		panic(fmt.Sprintf("There should be a map here for %s!\n", mbean))
+	if !ok {
+		return nil, fmt.Errorf("the response of %s's value should be a map", mbean)
 	}
 
 	points := make([]point, 0)
@@ -46,7 +46,7 @@ func (pb *pointBuilder) Build(mbean string, value interface{}) []point {
 		})
 	}
 
-	return compactPoints(points)
+	return compactPoints(points), nil
 }
 
 // extractTags generates the map of tags for a given mbean name/pattern.

